src/components/VoteCard.jsx

```diff
@@ -1,9 +1,37 @@
+import {useEffect, useState} from "react";
+import {patchArticleVotes} from "../utils/api";
+
 export default function VoteCard({votes, article_id}) {
+  const [newVotes, setNewVotes] = useState(votes);
+
+  const [errMsg, setErrMsg] = useState(undefined);
+  const voteUpHandler = (e) => {
+    updateArticleVote({inc_votes: 1});
+  };
+  const voteDownHandler = (e) => {
+    updateArticleVote({inc_votes: -1});
+  };
+  const updateArticleVote = (vote) => {
+    patchArticleVotes(article_id, vote)
+      .then(({article}) => {
+        setNewVotes((currVotes) => {
+          return currVotes + vote.inc_votes;
+        });
+      })
+      .catch((err) => {
+        setErrMsg(err.response.data.msg);
+      });
+  };
```

Comment:
We need to handle the voting optimistically. 
there's some good NCnotes on this. 
Currently we're having to wait for the api before it updates. If the server/my internet is slow, i could keep clicking the button thinking that it isnt working, then end up voting more times than expected.